### ISSUE TITLE
Self-check doesn't pass with "no last known location"

### ISSUE THREAD

**Issue by DocSniper** [microg/android_packages_apps_GmsCore#966]
As described in #964, the self-test fails even if I get a location.

![scnp](https://user-images.githubusercontent.com/3104188/67634065-604af080-f8af-11e9-8359-8af8e1998d4f.png)

Permissions don't seem to be the problem:
> runtime permissions:
> - android.permission.ACCESS_FINE_LOCATION: granted=true, flags=[ SYSTEM_FIXED GRANTED_BY_DEFAULT ]
> - android.permission.READ_EXTERNAL_STORAGE: granted=true, flags=[ GRANTED_BY_DEFAULT ]
> - android.permission.ACCESS_COARSE_LOCATION: granted=true, flags=[ SYSTEM_FIXED GRANTED_BY_DEFAULT ]
> - android.permission.READ_PHONE_STATE: granted=true, flags=[ GRANTED_BY_DEFAULT ]
> - android.permission.GET_ACCOUNTS: granted=true, flags=[ GRANTED_BY_DEFAULT ]
> - android.permission.WRITE_EXTERNAL_STORAGE: granted=true, flags=[ GRANTED_BY_DEFAULT ]

@Iey4iej3 recommended to test other NLP backends, but I installed almost all of them (except the Apple backend, which didn't solve the problem either).

![nlpbe](https://user-images.githubusercontent.com/3104188/67634064-604af080-f8af-11e9-85ec-14fb8e50718f.png)

**Comment by benyaminl** [comment]
> As described in #964, the self-test fails even if I get a location.
> 
> ![scnp](https://user-images.githubusercontent.com/3104188/67634065-604af080-f8af-11e9-8359-8af8e1998d4f.png)
> 
> Permissions don't seem to be the problem:
> 
> > runtime permissions:
> > 
> > * android.permission.ACCESS_FINE_LOCATION: granted=true, flags=[ SYSTEM_FIXED GRANTED_BY_DEFAULT ]
> > * android.permission.READ_EXTERNAL_STORAGE: granted=true, flags=[ GRANTED_BY_DEFAULT ]
> > * android.permission.ACCESS_COARSE_LOCATION: granted=true, flags=[ SYSTEM_FIXED GRANTED_BY_DEFAULT ]
> > * android.permission.READ_PHONE_STATE: granted=true, flags=[ GRANTED_BY_DEFAULT ]
> > * android.permission.GET_ACCOUNTS: granted=true, flags=[ GRANTED_BY_DEFAULT ]
> > * android.permission.WRITE_EXTERNAL_STORAGE: granted=true, flags=[ GRANTED_BY_DEFAULT ]
> 
> @Iey4iej3 recommended to test other NLP backends, but I installed almost all of them (except the Apple backend, which didn't solve the problem either).
> 
> ![nlpbe](https://user-images.githubusercontent.com/3104188/67634064-604af080-f8af-11e9-85ec-14fb8e50718f.png)

I also want to know what's the side effect of this as for everything like position have no problem for me on OSM and whatsapp. Is there maybe side effect. I also have similliar problem where UnifiedNlp not registered in system. Is there anything wrong? How to debug it?

**Comment by Iey4iej3** [comment]
@benyaminl Do you have "not registered" issue in the [latest build](https://github.com/microg/android_packages_apps_GmsCore/issues/964#issuecomment-546473162)?

**Comment by benyaminl** [comment]
> @benyaminl Do you have "not registered" issue in the [latest build](https://github.com/microg/android_packages_apps_GmsCore/issues/964#issuecomment-546473162)?

Seems it's solved, when I don't set it as system app(move to system) it will happen, now it's working. Thanks

**Comment by DocSniper** [comment]
@Iey4iej3 Maybe related to "your" issue 22 posted in lineageos4microg or it's even the same issue you reported there ?
https://github.com/lineageos4microg/android_prebuilts_prebuiltapks/issues/22

**Comment by Iey4iej3** [comment]
No. Just look at the screen shot: "UnifiedNlp is registered in system" in your case.

**Comment by DocSniper** [comment]
Alright, I see, I have to deepen my research.

**Comment by Iey4iej3** [comment]
I suggest you attach a logcat, at least grep with keywords `location`, `ichnaea` (if Mozilla is enabled).

**Comment by DocSniper** [comment]
I attached logs with the two filters set.
[logcat_ichnaea.txt](https://github.com/microg/android_packages_apps_GmsCore/files/3814751/logcat_ichnaea.txt)
[logcat_location.txt](https://github.com/microg/android_packages_apps_GmsCore/files/3814752/logcat_location.txt)

**Comment by Iey4iej3** [comment]
@DocSniper Which build are you using? I see that Qualcomm Location (iZat) is working in your system. Then it seems more difficult to analyze the situation. Wait for others to comment.
By the way, I suggest you to clean the privacy info in the log (say, latitude, longitude, mac).

**Comment by DocSniper** [comment]
My configuration is:
- LineageOS 16 (official) with LOS addonsu
- GmsCore 0.2.10.19420
- Phone: OnePlus 3T (SoC Qualcomm MSM8996 Pro / Snapdragon 821)

Thanks for the privacy hint, but the logs shouldn't contain real coordinates, cell ids, etc. 
I don't live @ 99,12345-99,12345 :grin:
